sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JsonInferSchema.scala

```diff
@@ -68,13 +71,15 @@ private[sql] class JsonInferSchema(options: JSONOptions) extends Serializable {
             Some(inferField(parser))
           }
         } catch {
-          case  e @ (_: RuntimeException | _: JsonProcessingException) => parseMode match {
-            case PermissiveMode =>
-              Some(StructType(Seq(StructField(columnNameOfCorruptRecord, StringType))))
-            case DropMalformedMode =>
-              None
-            case FailFastMode =>
-              throw QueryExecutionErrors.malformedRecordsDetectedInSchemaInferenceError(e)
+          case e @ (_: RuntimeException | _: JsonProcessingException |
+                    _: CharConversionException | _: MalformedInputException) =>
```

Comment:
Let's separate CharConverrsionException like JacksonParser, and change its cause like:
https://github.com/apache/spark/blob/186815be1c0832e2f976faf6f2d5c31e42eebf1a/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala#L495-L501

BTW, the indentation below here seems wrong

Reply:
cases like https://github.com/apache/spark/pull/33706/files#diff-587e4c03ef08aa4964d4409ae33954735a9c090727dd7b5a918892cd3ed0814dR2450 seems not to have a chance to add JSON options

Reply:
And when reading from files, `CharConversionException` can be not related to the value, but also something else like compression. Maybe we shall handle it differently from the JacksonParser above

Reply:
It actually does via rawParser, e.g.) https://github.com/apache/spark/blob/0494dc90af48ce7da0625485a4dc6917a244d580/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JsonDataSource.scala#L105-L111

Let's match it first and fix it in a different place.